zio-http/src/main/scala/zhttp/service/Client.scala

```diff
@@ -63,12 +63,12 @@ final case class Client[R](rtm: HttpRuntime[R], cf: JChannelFactory[JChannel], e
   ): JChannelFuture = {
 
     try {
-      val uri  = new URI(jReq.uri())
-      val host = if (uri.getHost == null) jReq.headers().get(HeaderNames.host) else uri.getHost
-
-      assert(host != null, "Host name is required")
-
-      val port = req.url.port.getOrElse(80)
+      val (host, port) = (req.url.host, req.url.port) match {
+        case (Some(host), Some(port)) => (host, port)
+        case (Some(host), None) => (host, 80)
+        case _ => assert(false, "Host name is required")
+          ("", -1)
+      }
```

Comment:
Let's avoid tuple allocations as much as possible.

Reply:
Is this for performance reasons or for style preferences?
I've changed the approach a bit, please take a look.